builtin/credential/approle/path_role.go

```diff
@@ -169,11 +175,96 @@ can only be set during role creation and once set, it can't be reset later.`,
 			},
 		},
 		ExistenceCheck: b.pathRoleExistenceCheck,
-		Callbacks: map[logical.Operation]framework.OperationFunc{
-			logical.CreateOperation: b.pathRoleCreateUpdate,
-			logical.UpdateOperation: b.pathRoleCreateUpdate,
-			logical.ReadOperation:   b.pathRoleRead,
-			logical.DeleteOperation: b.pathRoleDelete,
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.CreateOperation: &framework.PathOperation{
+				Callback:  b.pathRoleCreateUpdate,
```

Comment:
general question
so the callback function (can, doesn't always) returns a `logical.ErrorResponse` if there's an error

like would all `logical.ErrorResponse` return a 400 response, but then 403's would be handled higher in the chain

Reply:
It looks like the `sdk/logical.Response` struct is purely for the response body -- it does not include any status information. The status codes probably get attached elsewhere. Right now I am not translating any of the error statuses in the openapi spec so it's not particularly relevant 🤷